Civi/Financeextras/Hook/ValidateForm/MembershipCreate.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Civi\Financeextras\Hook\ValidateForm;
+
+use Civi\Financeextras\Utils\OptionValueUtils;
+
+class MembershipCreate {
+
+  /**
+   * @param \CRM_Member_Form_Membership $form
+   * @param array $fields
+   * @param array $errors
+   */
+  public function __construct(private \CRM_Member_Form_Membership $form, private array &$fields, private array &$errors) {
+  }
+
+  public function handle() {
+    $this->validatePaymentForm();
+  }
+
+  public function validatePaymentForm() {
+    if (empty($this->fields['record_contribution'])) {
+      // Don't do anything if user will not be recording contribution
+      return;
+    }
+
+    if (!empty($this->fields['payment_plan_schedule'])) {
```

Comment:
you should instead check on this field `contribution_type_toggle` if it is value != 'payment_plan'